File: Data Structures & Algorithms/two-integer-sum/submission-3.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> st;
        for(int i=0;i<nums.size();i++){
            if(st.find(nums[i])==st.end()){
                st[nums[i]] = i; 
            }
        }
        vector<int> ans;
        for(int i=0;i<nums.size();i++){
            auto it = st.find(target-nums[i]);
            if(st.find(target-nums[i])!=st.end() && i!=it->second){
                auto it = st.find(target-nums[i]);
                ans.push_back(i);
                ans.push_back(it->second); 
                break;
                         
            }
        }
        sort(ans.begin(),ans.end());
        return ans;
        
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:

        m = {}
        for i in nums:
            if i not in m:
                m[i] = 1
            else:
                m[i] += 1

        l = list(m.items())
        l = sorted(l, key=lambda x:x[1])
        n = len(l)
        ans = []
        for i in range(n-1,n-1-k,-1):
            ans.append(l[i][0])
        return ans



        

        